### PR TITLE
[SR SQL Planner Summer Camp] New Parser and Analyzer support LoadStmt

### LINKED ISSUE #7955: New Parser and Analyzer support Load related-statements (Challenging)
# Required work and output

1.  Use antlr4 to complete parse statements. The generated statement must follow the original data structure, and should be compatible with the old statement.
    
2.  In the original syntax file  **sql_parser.cup**, the new syntax needs to be compatible with the syntax of the old version, using the syntax file sql_parser.cup of the old version as the reference standard.
    
3.  Remove the logic of analytic functions in the statement and move the relevant logic to the new Analyzer framework. The new Analyzer is in the SQL/Analyzer directory.
    
4.  If the code related to Privilege Check exists in the original analyzer, move the code related to privilege control into PrivilegeChecker.
    
5.  When submitting the PR, please make sure there is unit test coverage for related changes.
    

# Tasks

 - [ ] ShowLoadStmt
 - [ ] CancelLoadStmt
 - [ ] ShowLoadWarningsStmt
 - [ ] LoadStmt
 - [ ] ModifyBrokerClause
 - [ ] ShowBrokerStmt

# How to participate?

1.  Pick an issue you are interested in, click the issue link and leave a comment under the issue to sign up.  **One person can only choose one issue and the first person who picks the issue will be assigned.**
    
2.  **After the issue is assigned, you should complete the task in 14 days.** If you are unable to complete it within the specified time, the task will be withdrawn and assigned to others.
    
3.  Participants are required to produce two pieces of content (forms including but not limited to mind maps, study notes, blogs, and tutorials) in the course of the activity, each of which should be at least 200 words.
    
4.  Got questions? Join our  [Slack channel](https://join.slack.com/t/starrocks/shared_invite/zt-192zeqlc7-9MAvqAXNcveW_l5Pqr8r2g)  and find channel:  **#sql-planner**.
    

# Before you start

1.  Read  [Contributing.md](https://github.com/StarRocks/starrocks/blob/main/CONTRIBUTING.md)  carefully to understand how to submit a pull request and how to configure BE and FE development environment.
    
2.  Sign the  [Contributor License Agreement](https://cla-assistant.io/StarRocks/starrocks)  before you make a pull request.
    
3.  To learn more about StarRocks, please refer to  [StarRocks documentation](https://github.com/StarRocks/docs/blob/main/TOC.md).
    

# Pull request requirements

1.  Create a pull request (PR) after you complete the task.
* PR title format: [SR SQL Planner Summer Camp] + PR title
2. After submitting PR, you can comment under the pinned Issue [SR SQL Planner Summer Camp].
* Comment format: "I have finished Issue #" + PR link

### PR DESCRIPTION
**Pull request by predator4ann**
## What type of PR is this：
- [x] feature

## Which issues of this PR fixes ：
Fixes #7955

## Problem Summary(Required) ：
New Parser and Analyzer support LoadStmt

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 131 / 160 (81.88%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[299]|
|:large_blue_circle:|com/starrocks/sql/analyzer/LoadStmtAnalyzer.java|55|69|79.71%|[25, 50, 63, 79, 83, 95, 96, 97, 98, 101, 115, 123, 124, 132]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|67|80|83.75%|[2069, 2102, 2134, 2135, 2136, 2137, 2138, 2150, 2151, 2152, 2153, 2154, 2158]|
|:large_blue_circle:|com/starrocks/analysis/LoadStmt.java|5|6|83.33%|[281]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/LabelName.java|2|2|100.00%|[]|

**Comment by mofeiatwork** [comment]
@predator4ann Please rebase the code to resolve conflicts.

File: fe/fe-core/src/test/java/com/starrocks/analysis/LoadStmtTest.java [hunk 1]
```diff
@@ -95,39 +66,57 @@ public void testNormal(@Injectable DataDescription desc, @Mocked GlobalStateMgr
...
+                "LOAD LABEL test.testLabel " +
+                        "(DATA INFILE(\"hdfs://hdfs_host:hdfs_port/user/starRocks/data/input/file\") INTO TABLE `t0`) " +
+                        "WITH RESOURCE spark0 " +
+                        "PROPERTIES (\"strict_mode\"=\"true\")");
         Assert.assertEquals(EtlJobType.SPARK, stmt.getResourceDesc().getEtlJobType());
-        Assert.assertEquals("LOAD LABEL `default_cluster:testDb`.`testLabel`\n(XXX)\nWITH RESOURCE 'spark0'",
-                stmt.toString());
+        Assert.assertEquals("root", stmt.getUser());
+        Assert.assertEquals(ImmutableSet.of("strict_mode"), stmt.getProperties().keySet());
+//        Assert.assertEquals("LOAD LABEL `default_cluster:testDb`.`testLabel`\n(XXX)\nWITH RESOURCE 'spark0'",
```
Comment (mofeiatwork) [inline fe/fe-core/src/test/java/com/starrocks/analysis/LoadStmtTest.java:94]: What's the meaning?

File: fe/fe-core/src/test/java/com/starrocks/analysis/LoadStmtTest.java [hunk 2]
```diff
@@ -21,72 +21,43 @@
...
     @Test
-    public void testNormal(@Injectable DataDescription desc, @Mocked GlobalStateMgr globalStateMgr,
-                           @Injectable ResourceMgr resourceMgr, @Injectable Auth auth)
-            throws UserException, AnalysisException {
-        List<DataDescription> dataDescriptionList = Lists.newArrayList();
-        dataDescriptionList.add(desc);
+    public void testNormal(@Injectable DataDescription desc, @Mocked GlobalStateMgr globalStateMgr, @Injectable ResourceMgr resourceMgr, @Injectable
+    Auth auth)
+            throws UserException {
+//        resourceMgr.createResource(new CreateResourceStmt(true, "spark0", Maps.newHashMap()));
```
Comment (mofeiatwork) [inline fe/fe-core/src/test/java/com/starrocks/analysis/LoadStmtTest.java:55]: ?

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/LoadStmtAnalyzer.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        private void analyzeLabel(LoadStmt statement, ConnectContext context) {
+            LabelName label = statement.getLabel();
+            String dbName = label.getDbName();
+            if (Strings.isNullOrEmpty(dbName)) {
+                dbName = context.getDatabase();
+                if (Strings.isNullOrEmpty(dbName)) {
+                    ErrorReport.reportSemanticException(ErrorCode.ERR_NO_DB_ERROR);
+                }
+            }
+            dbName = ClusterNamespace.getFullName(dbName);
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/LoadStmtAnalyzer.java:53]: Check db existence here?

**Comment by predator4ann** [comment]
@mofeiatwork rebase done

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 136 / 167 (81.44%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[303]|
|:large_blue_circle:|com/starrocks/analysis/LoadStmt.java|5|7|71.43%|[255, 277]|
|:large_blue_circle:|com/starrocks/sql/analyzer/LoadStmtAnalyzer.java|60|75|80.00%|[56, 63, 64, 74, 90, 94, 107, 108, 109, 110, 113, 127, 135, 136, 144]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|67|80|83.75%|[2226, 2259, 2291, 2292, 2293, 2294, 2295, 2307, 2308, 2309, 2310, 2311, 2315]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/LabelName.java|2|2|100.00%|[]|

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run starrocks_clang-format

**Review by murphyatwork — DISMISSED** [review]

**Review by kangkaisen — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 136 / 167 (81.44%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[304]|
|:large_blue_circle:|com/starrocks/analysis/LoadStmt.java|5|7|71.43%|[255, 277]|
|:large_blue_circle:|com/starrocks/sql/analyzer/LoadStmtAnalyzer.java|60|75|80.00%|[56, 63, 64, 74, 90, 94, 107, 108, 109, 110, 113, 127, 135, 136, 144]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|67|80|83.75%|[2227, 2260, 2292, 2293, 2294, 2295, 2296, 2308, 2309, 2310, 2311, 2312, 2316]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/LabelName.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by predator4ann** [comment]
run starrocks_fe_unittest

**Comment by predator4ann** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 135 / 166 (81.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[304]|
|:large_blue_circle:|com/starrocks/analysis/LoadStmt.java|5|7|71.43%|[255, 277]|
|:large_blue_circle:|com/starrocks/sql/analyzer/LoadStmtAnalyzer.java|59|74|79.73%|[57, 64, 65, 75, 91, 95, 108, 109, 110, 111, 114, 128, 136, 137, 145]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|67|80|83.75%|[2207, 2240, 2272, 2273, 2274, 2275, 2276, 2288, 2289, 2290, 2291, 2292, 2296]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/LabelName.java|2|2|100.00%|[]|

**Review by murphyatwork — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**